src/test/java/org/isf/patient/test/Tests.java

```diff
@@ -232,6 +278,448 @@ public void testMergePatientHistory() throws Exception {
+		assertThat(patientBrowserManager.getProfessionTranslated("mining")).isEqualTo("angal.patient.profession.mining");
+	}
+
+	@Test
+	public void testMgrGetProfessionKey() throws Exception {
+		resetHashMaps();
+		// TODO: if resource bundles are made avaiable in core then the values being compared will need to change
+		assertThat(patientBrowserManager.getProfessionKey(null)).isEqualTo("undefined");
+		assertThat(patientBrowserManager.getProfessionKey("someKeyNotInTheList")).isEqualTo("undefined");
+		assertThat(patientBrowserManager.getProfessionKey("angal.patient.profession.mining")).isEqualTo("mining");
+	}
+
+	@Test
+	public void testMgrPatientValidationNoFirstName() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setFirstName("");
+
+			patientBrowserManager.savePatient(patient);
+		})
+				.isInstanceOf(OHServiceException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error"));
+	}
+
+	@Test
+	public void testMgrPatientValidationNullFirstName() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setFirstName(null);
+
+			patientBrowserManager.savePatient(patient);
+		})
+				.isInstanceOf(OHServiceException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error"));
+	}
+
+	@Test
+	public void testMgrPatientValidationNoSecondName() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setSecondName("");
+
+			patientBrowserManager.savePatient(patient);
+		})
+				.isInstanceOf(OHServiceException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error"));
+	}
+
+	@Test
+	public void testMgrPatientValidationNullSecondName() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setSecondName(null);
+
+			patientBrowserManager.savePatient(patient);
+		})
+				.isInstanceOf(OHServiceException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error"));
+	}
+
+	@Test
+	public void testMgrPatientValidationBirthDateNull() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setBirthDate(null);
+
+			patientBrowserManager.savePatient(patient);
+		})
+				.isInstanceOf(OHServiceException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error"));
+	}
+
+	@Test
+	public void testMgrPatientValidationBirthDateTooFarInFuture() {
+		assertThatThrownBy(() -> {
+			Patient patient = testPatient.setup(true);
+
+			patient.setBirthDate(new Date(9999, 1, 1));
```

Comment:
using the deprecated constructor Date(int, int, int) is part of the tested exception?

Reply:
No; I was just lazy and it was easier that way.

Reply:
Thanks!